File: src/components/StopHeader.ts
// StopHeader Component
// Generates the header section for a stop page
// Using as-bind for proper WASM memory management

/**
 * Generates the header for a stop page
 */
export function StopHeader(
  stopName: string,
  stopId: string,
  parentStopId: string,
  parentStopName: string,
  agencyId: string
): string {
  let html = '<h1>' + stopName + '</h1>';
  
  // If this is a child stop, show link to parent
  if (parentStopId !== '' && parentStopName !== '') {
    html += '<p><a href="/' + agencyId + '/stops/' + parentStopId + '/">← Back to ' + parentStopName + '</a></p>';
  }
  
  html += '  <p>Stop ID: ' + stopId + '</p>';
  
  return html;
}

File: rust_src/components/stop_header.rs
// Stop Header Component

pub fn render(
    stop_name: &str,
    stop_id: &str,
    parent_id: &str,
    parent_name: &str,
    _agency_id: &str,
) -> String {
    let mut html = format!("  <h1>{}</h1>\n", stop_name);
    html.push_str(&format!("  <p>Stop ID: {}</p>\n", stop_id));
    
    if !parent_id.is_empty() {
        html.push_str(&format!(
            "  <p><a href=\"/stops/{}/\">← Back to {}</a></p>\n",
            parent_id, parent_name
        ));
    }
    
    html
}
